src/client/datascience/notebook/helpers/helpers.ts

```diff
@@ -55,42 +54,41 @@ export function isJupyterNotebook(option: NotebookDocument | string) {
     }
 }
 
-const kernelInformationForNotebooks = new WeakMap<
-    NotebookDocument,
-    { metadata?: KernelConnectionMetadata | undefined; kernelInfo?: Partial<KernelMessage.IInfoReplyMsg['content']> }
->();
-
 export function isResourceNativeNotebook(resource: Resource, notebooks: IVSCodeNotebook, fs: IFileSystem) {
     if (!resource) {
         return false;
     }
     return notebooks.notebookDocuments.some((item) => fs.arePathsSame(item.uri, resource));
 }
-export function getNotebookMetadata(document: NotebookDocument): nbformat.INotebookMetadata | undefined {
+export function getNotebookMetadata(document: NotebookDocument | NotebookData): nbformat.INotebookMetadata | undefined {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let notebookContent: Partial<nbformat.INotebookContent> = document.metadata.custom as any;
-
-    // If language isn't specified in the metadata, at least specify that
-    if (!notebookContent?.metadata?.language_info?.name) {
-        const content = notebookContent || {};
-        const metadata = content.metadata || { orig_nbformat: defaultNotebookFormat.major, language_info: {} };
-        const language_info = { ...metadata.language_info };
-        // Fix nyc compiler not working.
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        notebookContent = { ...content, metadata: { ...metadata, language_info } } as any;
-    }
-    notebookContent = cloneDeep(notebookContent);
-    const data = kernelInformationForNotebooks.get(document);
-    if (data && data.metadata) {
-        updateNotebookMetadata(notebookContent.metadata, data.metadata, data.kernelInfo);
-    }
+    const notebookContent: undefined | Partial<nbformat.INotebookContent> = document.metadata?.custom as any;
+    // Create a clone.
+    return JSON.parse(JSON.stringify(notebookContent?.metadata || {}));
```

Comment:
I believe that we use lodash cloneDeep in other places. Advantage to one over the other? Looking it up now.

Reply:
Sounds like this is fine when only dealing with simple types, so fine to leave here as is, was just interested.

Reply:
I think this is faster, as we do'nt have any non-primitive types in the JSON (we don't have functions, dates, etc).